### PR TITLE
Refactor rules: align rules

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Changed:
- Refactored align/*.rules to make it more readable.
- Picard's MarkDuplicate is now part of bwa_mem rule

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/snakemake_rules/align/sentieon_alignment.rule [hunk 1]
```diff
@@ -75,18 +79,18 @@ rule sentieon_realign:
...
-        sentieon_exec = config["SENTIEON_INSTALL_DIR"] + "/bin/sentieon",
+        sentieon_exec = config["SENTIEON_EXEC"],
         sentieon_lic = config["SENTIEON_LICENSE"],
     threads: get_threads(cluster_config, 'sentieon_realign')
     log:
-        bam_dir + "{sample}.dedup.realign.bam.log"
+        Path(bam_dir, "{sample}.dedup.realign.bam.log").as_posix()
     benchmark:
-        benchmark_dir + 'sentieon_realign_' + "{sample}.dedup_realign.tsv"
+        Path(benchmark_dir + "sentieon_realign_{sample}.dedup_realign.tsv").as_posix()
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/align/sentieon_alignment.rule:93]:
```suggestion
        Path(benchmark_dir, "sentieon_realign_{sample}.dedup_realign.tsv").as_posix()
```

File: BALSAMIC/snakemake_rules/align/sentieon_alignment.rule [hunk 2]
```diff
@@ -106,20 +110,20 @@ rule sentieon_base_calibration:
         mills = config["reference"]["mills_1kg"],
         indel_1kg = config["reference"]["1kg_known_indel"],
         dbsnp = config["reference"]["dbsnp"],
-        bam = bam_dir + "{sample}.dedup.realign.bam",
+        bam = Path(bam_dir, "{sample}.dedup.realign.bam").as_posix()
     output:
         recal_data_table = bam_dir + "{sample}.dedup.realign.recal_data.table",
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/align/sentieon_alignment.rule:115]:
```suggestion
        recal_data_table = Path(bam_dir, "{sample}.dedup.realign.recal_data.table").as_posix(),
```

File: BALSAMIC/snakemake_rules/align/sentieon_alignment.rule [hunk 3]
```diff
@@ -106,20 +110,20 @@ rule sentieon_base_calibration:
...
         qual_recal = bam_dir + "{sample}.dedup.realign.recal.csv",
         qual_recal_plot = bam_dir + "{sample}.dedup.realign.recal.pdf",
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/align/sentieon_alignment.rule:117]: Change to Path

**Review by ashwini06 — COMMENTED** [review]
Shifting to `Path` is not fully implemented for all inputs and outputs in the rules.

File: BALSAMIC/snakemake_rules/align/sentieon_alignment.rule (hunk 3 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/align/sentieon_alignment.rule]: fixed!

File: BALSAMIC/snakemake_rules/align/sentieon_alignment.rule (hunk 2 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/align/sentieon_alignment.rule]: fixed!

File: BALSAMIC/snakemake_rules/align/sentieon_alignment.rule (hunk 1 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/align/sentieon_alignment.rule]: fixed!

**Comment by hassanfa** [comment]
@ashwini06 `git diff` picard Markduplicate is misleading. What happened is that I moved the rule from `picard.rule` to `bwa_mem.rule`. The idea is to have all align related rules in one place. This is to be able to switch to bwa mem for WGS if sentieon is not available.

**Review by ashwini06 — COMMENTED** [review]
👍

**Pull request merged**